lib/index.js

```diff
@@ -121,12 +121,19 @@ function Socket (type, opts) {
   }
 
   /* start listening for inbound messages */
-  if(this.af_domain === nn.AF_SP) {
-    if (this.receiver) this._startPollReceive();
+  if (this.af_domain === nn.AF_SP && this.receiver) {
+    this._startPollReceive();
   }
+
+  if (!opts.hasOwnProperty('objectMode')) {
+    opts.objectMode = true;
+  }
```

Comment:
it reduces the highwatermark from 16K msgs to 16msgs, it's much more useful if we have a backpressure mechanism